src/IPCMessenger.ts

```diff
@@ -0,0 +1,39 @@
+type Room = string;
+type Instance = string;
+
+enum MessageTypes {
+  Handover = 'handover',
+  Leave = 'leave'
+}
+
+type HandoverMessage = {
+  type: MessageTypes.Handover,
+  sender: Instance,
+  state?: Record<string, unknown>,
+};
+
+type LeaveMessage = {
+  type: MessageTypes.Leave,
+  sender: Instance,
+};
+
+type Message = HandoverMessage | LeaveMessage;
+
+type MessageCallback = (message: Message) => void;
+
+interface IPCMessenger {
+  join(room: Room, callback: MessageCallback): Promise<void>;
+  getOtherInstances(room: Room): Promise<Array<Instance>>;
+  send(room: Room, message: Message): Promise<void>;
+}
+
+export default IPCMessenger;
+export {
```

Comment:
This is unnecessary - just export when declaring the things. These are just extra lines of code that don't do anything.

Reply:
We have already adopted this approach eveywhere else, probably because it offers a cleaner separation of exports and logic.

Reply:
Nah, this is a nodejs-specific JS leftover, as node doesn't support `export` without TS. The introduction of `export` is considered a great improvement, as you immediately see whether a declaration is exported or not.

Reply:
I meant when looking at the exported code.

Reply:
I don't understand. Both of these are equivalent:

```js
export const someFunction = () => {};

export {
  someFunction
}
```

They both create an object with exports. In the first case you don't have to look in two places to know whether a function is exported.

Reply:
Its more evident when you have a lot of stuff.

```
const someFunction = () => {};
const someOtherFunction = () => {};
const theSameFunction = () => {};

/* ... */
/* ... */
/* ... */

export {
  someFunction
}
```

Reply:
Well, you shouldn't have a lot of stuff in general, but especially when you have a lot of stuff it's much easier to identify what's exported and what isn't when it says so right where the thing is.

Here's an analogy with the private/public modifiers on a class level - imagine that you would declare your fields at the top of the class, and then declare which one is private/public at the bottom of the file. Does that seem logical?

Reply:
> Here's an analogy with the private/public modifiers on a class level - imagine that you would declare your fields at the top of the class, and then declare which one is private/public at the bottom of the file. Does that seem logical?

In that case, it is also illogical to mix them throughout the whole file. IMO, it's cleaner that way, but let's discuss with the whole team.